crates/ethcore/src/client/traits.rs

```diff
@@ -437,6 +444,66 @@ pub trait BlockChainClient:
     fn is_processing_fork(&self) -> bool;
 }
 
+/// The data required for a `Client` to create a transaction.
+///
+/// Gas limit, gas price, or nonce can be set explicitly, e.g. to create service
+/// transactions with zero gas price, or sequences of transactions with consecutive nonces.
+/// Added for AuRa needs.
+pub struct TransactionRequest {
```

Comment:
this is helper struct used for two functions above, it can have optional fields and it is not compatible with `common_types::transaction::Transaction`